polyval/src/field/backend/soft.rs

```diff
@@ -1,18 +1,45 @@
-//! Software emulation support for CLMUL hardware intrinsics.
-//!
-//! WARNING: Not constant time! Should be made constant-time or disabled by default.
-
-// TODO(tarcieri): performance-oriented constant-time implementation
-// See: <https://bearssl.org/gitweb/?p=BearSSL;a=blob;f=src/hash/ghash_ctmul64.c>
+//! Constant-time software implementation of POLYVAL
+
+// Adapted from BearSSL's `ghash_ctmul64.c`
+// <https://bearssl.org/gitweb/?p=BearSSL;a=blob;f=src/hash/ghash_ctmul64.c;hb=4b6046412bf927d6424f20fc7ee495bb96dbd227>
+//
+// Copyright (c) 2016 Thomas Pornin <pornin@bolet.org>
```

Comment:
I asked @pornin about licensing and he says it's fine to relicense under `MIT OR Apache-2.0`. I'm not sure how we want to reflect that in the comments. Perhaps I can just delete the text below?

Reply:
Yeah, I think it's fine to remove this text.